src/ml/neural_net/style_transfer/mps_style_transfer.m

```diff
@@ -251,6 +271,60 @@ - (instancetype) initWithDev:(id<MTLDevice>) dev
   return self;
 }
 
++ (NSData *) cropImageWidth:(NSData *) data
+                     height:(NSUInteger) height
+                 inputWidth:(NSUInteger) inputWidth
+                outputWidth:(NSUInteger) outputWidth
+                numChannels:(NSUInteger) numChannels {
+  NSUInteger outputLength = height * outputWidth * numChannels;
+  NSMutableData *mutableData = [NSMutableData dataWithLength:outputLength];
+
+  NSUInteger xOffset = inputWidth - outputWidth;
+
+  for (NSUInteger idx = 0; idx < height; idx = idx + 1) {
+    NSUInteger startIndex = idx * inputWidth * numChannels;
+    NSUInteger endIndex = startIndex + (numChannels * (inputWidth - xOffset));
+
+    NSRange range = NSMakeRange(startIndex, endIndex);
```

Comment:
There should be a `sizeof(float)` in here somewhere, no?

Reply:
yeah, I ran into issues testing it, I added size of float here.